### DISCUSSION TITLE
Java agent collector config with gRPC URL

### DISCUSSION THREAD

**Discussion by mattwmj** [apache/skywalking#9179] [Q&A]
Hello fellows,

I understand that we can provider java agent with config collector.backend_service to specify the backend service host.
for example:
collector.backend_service=oap.somehost.com:11800

I wonder if there is a way to provider gRPC url instead like:
collector.backend_service=grpc://somehost.com:11800/oap

Cheers!

**Comment by wu-sheng** [comment]
I don't think we have that.

**Comment by Superskyyy** [comment]
May I ask what is the purpose of using the gRPC URL scheme? Just curious to see the use case.

**Reply by mattwmj** [reply]
The system I am currently working on have domain name limitition and I want to avoid hard coded IP in agent config. 

So, Skywalking OAP cannot have its own domain name like 
oap.my-application.mycompany.com

I would like to have this grpc url feature combined with nginx grpc proxy so that Java agents can send data to nginx by url:
grpc://my-application.mycompany.com/oap
and nginx will proxy data to actual oap server IP. FYI: https://www.nginx.com/blog/nginx-1-13-10-grpc/

I also wonder if this is a valid requirement.

**Reply by wu-sheng** [reply]
Are you asking a grpc namespace?

**Reply by mattwmj** [reply]
> grpc namespace

I don't quite understand what a grpc namespace is.

I think maybe we can add a property with name 'collector.backend_service_url' in agent.config.
Default value of collector.backend_service_url will be grpc://${collector.backend_service} and I can change it to 
grpc://my-application.mycompany.com/oap.
And the Java gRPC client in agent can use collector.backend_service_url to send data request.

What do you think?

**Reply by wu-sheng** [reply]
If you don't understand gRPC, why do you ask us to support an unclear thing? I am confused.

**Reply by mattwmj** [reply]
I am not familiar with gRPC. I am looking for help...
I will do a little more study on gRPC.
